lib/Frontend/ModuleInterfaceSupport.cpp

```diff
@@ -351,7 +359,11 @@ static void printImports(raw_ostream &out,
         out << "@_spi(" << spiName << ") ";
     }
 
-    if (ctx.LangOpts.hasFeature(Feature::InternalImportsByDefault)) {
+    if (!Opts.printPublicInterface() &&
+        Opts.printPackageInterface() &&
+        packageOnlyImportSet.count(import))
+      out << "package ";
+    else if (ctx.LangOpts.hasFeature(Feature::InternalImportsByDefault)) {
```

Comment:
Would you mind adding a test where a module is imported from two files, once `public` and once `package`? Since we prioritize the `package` case here I'm worried we may print it out as a package import instead of a public one.

Reply:
Added a check L371 `!publicImportSet.count(import)`